src/System.Management.Automation/engine/CommandProcessorBase.cs

```diff
@@ -452,13 +458,16 @@ internal void DoPrepare(IDictionary psDefaultParameterValues)
                     HandleObsoleteCommand(ObsoleteAttribute);
                 }
             }
-            catch (Exception)
+            catch (Exception e)
             {
-                if (_useLocalScope)
+                if (e is InvalidComObjectException)
```

Comment:
Make sense to use `catch-when` since we re-throw others? I see below such patterns too.

Reply:
Yeah, actually after the change, we should replace `catch (Exception e)` with `catch (InvalidComObjectException e)` instead. Updated.